Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -69,18 +95,32 @@ public class HTTPNetworkTransport: NetworkTransport {
   ///
   /// - Parameters:
   ///   - operation: The operation to send.
+  ///   - fetchHTTPMethod: The HTTP Method to be used.
   ///   - completionHandler: A closure to call when a request completes.
   ///   - response: The response received from the server, or `nil` if an error occurred.
   ///   - error: An error that indicates why a request failed, or `nil` if the request was succesful.
   /// - Returns: An object that can be used to cancel an in progress request.
-  public func send<Operation>(operation: Operation, completionHandler: @escaping (_ response: GraphQLResponse<Operation>?, _ error: Error?) -> Void) -> Cancellable {
+  public func send<Operation>(operation: Operation, fetchHTTPMethod: FetchHTTPMethod, completionHandler: @escaping (_ response: GraphQLResponse<Operation>?, _ error: Error?) -> Void) -> Cancellable {
+    let body = requestBody(for: operation)
     var request = URLRequest(url: url)
-    request.httpMethod = "POST"
     
+    switch fetchHTTPMethod {
+    case .GET:
+      if let urlForGet = mountUrlWithQueryParamsIfNeeded(body: body) {
+        request = URLRequest(url: urlForGet)
+      } else {
+        completionHandler(nil, GraphQLHTTPRequestError(kind: .serializedQueryParamsMessageError))
+      }
+    default:
+      do {
+        request.httpBody = try serializationFormat.serialize(value: body)
+      } catch {
+        completionHandler(nil, GraphQLHTTPRequestError(kind: .serializedBodyMessageError))
```

Comment:
You should probably return here and above to prevent the rest of this from executing

Reply:
Hi! 

This method `send<Operation>` should return a `Cancellable`. In this case, it should return a task after the request. If it fails, it will execute the request and returns another error. 

As the task must resume ( `task.resume()` ), what do you think of instead of `do/try`, we use a `try?` to let the request executes and returns the correct error? Another option would be change the return type to be optional as `Cancellable?`. But I don't like that much this approach because it would change the whole architecture of `apollo-client` and also, the request would never be executed in this case. And I'm not sure that it would be a good idea for clients.

What do you think? Do you have any suggestion?

Reply:
I'll poke around - I think there's probably something else that conforms to `Cancelable` that we can return here

Reply:
Okay! Thank you, once again!

Reply:
Right now the only non private object/struct is an extension of `URLSessionTask` declared inside this class. The others are the classes `GraphQLQueryWatcher` and `WebSocketTask` which are not supposed to be used in here. 

But I was thinking about it... A request can be executed without a request body or a query parameter. It's like a request to an endpoint without parameters. So, if this try fails, it won't have a request body, and so, when the task resume, it will just return an error and the reason for it.

But also, the only way for this to fail, right now, would be in the case that `Operation` fails to be constructed. Which means a bad format in query that would fail while compiling or maybe during the processing. 

Maybe, that's why it was using `try!` before. I do not like to force a try, but changing it for `try?` wouldn't cause a crash. And because we are confident that GraphQLOperation only exists if it succeeded parsing those values in query, the worst case scenario would be a request to an endpoint without parameter.

Reply:
But once again, tell me what is your opinion, please! I want to help with it, but if you want to keep this change and find another way to solve it, please, be my guest to fix it.

Reply:
I think for this PR we're gonna leave it - I'll make a fix that addresses this in other areas as well in a separate PR.

Reply:
Ok, thank you for the review. :) !!